llava/VLLMSafety/discriminator.py

```diff
@@ -6,58 +6,95 @@
 import json
 import random
 
+
 class Discriminator(nn.Module):
-    def __init__(self, input_size):
-        super().__init__()  # we can add more layers later
+    def __init__(self, input_size, learning_rate=0.0002, adam_beta1=0.9):
+        super().__init__()
 
         self.fc1 = nn.Linear(input_size, 50)
         self.fc2 = nn.Linear(50, 1)
 
+        self.device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+        self.to(self.device)
+
     def linear(self, x):
         x = F.relu(self.fc1(x))
         x = torch.sigmoid(self.fc2(x))
 
         return x
- 
-    def forward(self, data, d_mode):
-        device = 'cuda'  
-        loss_function = nn.BCELoss() # from DCGAN
 
-        img_tok = data["image"]
-        lang_tok = data["lang"]
+    def run_forward(self, data, d_mode):
+
```

Comment:
can add some logic here to ensure that we send the same number of language and image tokens too to balance the set (typically more image data than lang)